sherpa/astro/flux.py

```diff
@@ -529,12 +529,91 @@ def sample_flux(fit, data, src,
 
 def calc_sample_flux(id, lo, hi, session, fit, data, samples, modelcomponent,
                      confidence):
+    """Given a set of parameter samples, estimate the flux distribution.
+
+    This is similar to sample_flux but returns values for both the full
+    and component models (which can be the same). The set of parameter
+    samples is searched to remove rows where the parameters lie outside
+    the soft limits, and then this set of rows is used to calculate the
+    flux of the modelcomponent expression (the flux for the full model
+    is taken from the samples argument). The resulting set of fluxes is
+    used to calculate the median and the confidence range.
+
+    .. versionchanged:: 4.13.1
+       The clipping now includes parameters at the soft limits;
+       previously they were excluded which could cause problems with
+       parameters for which we can only calculate an upper limit.
+
+    Parameters
+    ----------
+    id : int or str
+        The dataset identifier. It must exist and have an associated model in
+        session.
+    lo : number or None, optional
+        The lower edge of the dataspace range for the flux calculation.
+        If None then the lower edge of the data grid is used.
+    hi : number or None, optional
+        The upper edge of the dataspace range for the flux calculation.
+        If None then the upper edge of the data grid is used.
+    session : sherpa.ui.utils.Session instance
+        Defines the data, model, and fit.
+    fit : sherpa.fit.Fit instance
+        The fit object. The src parameter is assumed to be a subset of
+        the fit.model expression (to allow for calculating the flux of
+        a model component), and the fit.data object matches the data
+        object. The fit.model argument is expected to include instrumental
+        models (for PHA data sets). These objects can represent
+        simultaneous fits (e.g. sherpa.data.DataSimulFit and
+        sherpa.models.model.SimulFitModel).
+    data : sherpa.data.Data subclass
+        The data object to use. This is not a DataSimulFit instance.
+    samples
+        The output of sample_flux for the data (assumed to have been
+        created with method=calc_energy_flux and clip='hard').
+    modelcomponent : sherpa.models.Arithmetic instance
+        The source model (without instrument response for PHA data) that
+        is used for calculating the flux. This is not a SimulFitModel
+        instance.
+    confidence : number
+        The confidence value to return, as a percentage (0-100).
+
+    Returns
+    -------
+    (fullflux, cptflux, vals)
+       The fullflux and cptflux arrays contain the results for
+       the full source model and the flux of the `modelcomponent`
+       argument (they can be the same). They have three elements
+       and give the median value, upper quartile, and lower
```

Comment:
At least in the current version, the code only returns quartiles, if I set `confidence=50`, with e.g. `confidence=90` I get the 5-95% range, not the quartiles.

Reply:
This is a doc error - and is also present in the more-likely-to-be-used `sherpa.astro.ui.sample_flux` docstring (from where this was shamelessly copied) 

https://github.com/sherpa/sherpa/blob/b7cb202c27951c6dc2ee2ef0a2b661f1ff80c0db/sherpa/astro/ui/utils.py#L13317-L13329

Reply:
Can I ask you to fix the doc in `sample_flux` in this PR as well, since we now found it?

Reply:
Yeah, I just need the time.

Reply:
I've made a change. I'm not proud of the change, but I've made it.